src/NullConnection.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ *
+ * This file is part of Atlas for PHP.
+ *
+ * @license http://opensource.org/licenses/mit-license.php MIT
+ *
+ */
+declare(strict_types=1);
+
+namespace Atlas\Query;
+
+use Atlas\Pdo\Connection;
+use BadMethodCallException;
+
+class NullConnection extends Connection
+{
+    protected $driverName;
+
+    public static function new(...$args) : Connection
+    {
+        return new static(...$args);
+    }
+
+    public function __construct(string $driverName = 'null')
```

Comment:
This seems like a weak contract because the caller has no indication of what's valid here. I suggest accepting a Quoter instead, since that seems to be what the driver name is needed for.